mutiny-core/src/nodemanager.rs

```diff
@@ -1052,14 +1055,115 @@ impl<S: MutinyStorage> NodeManager<S> {
             return Err(MutinyError::WalletOperationFailed);
         };
 
+        // If we are in safe mode, we don't create payjoin sessions
+        let pj = if self.safe_mode {
+            None
+        } else {
+            let context = payjoin::receive::EnrollContext::from_relay_config(
+                "http://localhost:8080",
+                "AQAg7YjKSn1zBziW3LvPCQ8X18hH0dU67G-vOcMHu0-m81AABAABAAM",
+                "http://localhost:8080", // Ignore and pass directly to relay for now
+                                         //Some("c53989e590b0f02edeec42a9c43fd1e4e960aec243bb1e6064324bd2c08ec498")
+            );
+            let pj_uri = format!("{}{}", "http://localhost:8080/", context.subdirectory());
+            log_debug!(self.logger, "{pj_uri}");
+            let wallet = self.wallet.clone();
+            // run await payjoin task in the background as it'll keep polling the relay
+            wasm_bindgen_futures::spawn_local(async move {
+                let wallet = wallet.clone();
+                let pj_txid = Self::receive_payjoin(wallet, context).await.unwrap();
+                log::info!("Received payjoin txid: {}", pj_txid);
+            });
+            Some(pj_uri)
+        };
+
         Ok(MutinyBip21RawMaterials {
             address,
             invoice,
             btc_amount: amount.map(|amount| bitcoin::Amount::from_sat(amount).to_btc().to_string()),
             labels,
+            pj,
```

Comment:
I wonder if we should limit the options or allow user to specify a specific one. My concern is mostly showing both on chain address + lightning + payjoin in each QR code might make it pretty dense. Do you have an example payjoin uri with the new fields added for `ohttp` to get an idea of the size?

Reply:
It's an base64-url encoded ~128 bit pubkey + a few bytes to specify ciphersuite from a pre-defined list. We could (should?) encode &pj= and &ohttp= using the bitcoincommons UR encoding + Caps in the address, params, & BOLT 11 for a significantly more compact QR.

`bitcoin:tb1pr3hpfv99fhftn5gdxucgy4rax5mqggu2xc96puxgw5u4wp93tnrqgpcpmx?amount=0.00066588&lightning=lntbs665880n1pj5zdq4sp5n594wqz7cxfyxm8ngm6pjwt5x8ne9mjfjpejs8f9aq2jk42gdafqpp5jfjssftnn2un8e7wv94gp4e6zg65vpcycqcra3a6ccqgeh2myaxqdqqnp4qdn2hj8tfknpuvdg6tz9yrf3e27ltrx9y58c24jh89lnm43yjwfc5xqrrsxcqzzn9qyysgqumnvl5s6xg9hpcl4mvf8qfl4l6axyqj9vqaqrgflc2al690yxr5kpuyut6dfckpvts8fgx3uhqhat5zezwdzxgvjytmavkxsxe2cxkcqd88wvj&pj=http://localhost:8080/A_-kmO0WUPULpAWTCqt5J4IlbdynoTY0NsxkRBtKupoR&ohttp=AQAgwMcjpR7rer1o726dZj4n4-gqCdIVN7Q2bA6gZYwK2CwABAABAAM`